### ISSUE TITLE
Unpublished content is expected in Gatsby Cloud unless rebuilt without cache

### ISSUE THREAD

**Issue by josephmarkus** [contentstack/gatsby-source-contentstack#97]
Step to reproduce:

1. Unpublish an already published entry
2. Observe Gatsby Cloud failing

```
14:40:29 PM:
success write out requires - 0.018s

14:41:45 PM:
success Building production JavaScript and CSS bundles - 73.016s

14:42:09 PM:
success Building HTML renderer - 19.204s

14:42:26 PM:
failed Building static HTML for pages - 16.390s

14:42:26 PM:
ERROR Building static HTML failed for path "/some-page-from-content-stack"

14:42:26 PM:
  - static-entry.js:263 

14:42:28 PM:
ERROR Failed to compile: Error: Exited with code 1
```

Workaround:
1. Trigger build without cache in Gatsby Cloud

**Comment by vkalta** [comment]
@jacobimpson Unable to reproduce this. Can you please share the exact steps?

**Comment by josephmarkus** [comment]
@vkalta I assume the typo in your message :)

Step to reproduce:
1. Publish an entry in Contenstack (e.g. `"/some-page-from-content-stack"`)
2. Run Gatsby Cloud build that will fetch the published entry from Contenstack
3. Now that the entry is published and live go to Contenstack and unpublish the entry
4. Run Gatsby Cloud build
5. Observe an error where unpublished content is expected by Gatsby Cloud (`ERROR Building static HTML failed for path "/some-page-from-content-stack"`)

**Comment by vkalta** [comment]
> @vkalta I assume the typo in your message :)
> 
> Step to reproduce:
> 
> 1. Publish an entry in Contenstack (e.g. `"/some-page-from-content-stack"`)
> 2. Run Gatsby Cloud build that will fetch the published entry from Contenstack
> 3. Now that the entry is published and live go to Contenstack and unpublish the entry
> 4. Run Gatsby Cloud build
> 5. Observe an error where unpublished content is expected by Gatsby Cloud (`ERROR Building static HTML failed for path "/some-page-from-content-stack"`)

@josephmarkus Sorry about the typo. :). Will try these steps out and get back to you if I will need more details to debug this issue.

Thanks.

**Comment by vkalta** [comment]
@josephmarkus I am still not able to reproduce this. Gatsby cloud build works fine for the stack I am using.
Can you please share your stack details with me at vikram.kalta@contentstack.com?

Thanks

**Comment by kramerkm** [comment]
We're running into a similar issue, except we're using incremental builds with CircleCI. Whenever we unpublish an entry in Contentstack and run a build, it builds successfully but the page itself still exists on our website even though it was unpublished.

**Comment by vkalta** [comment]
@kramerkm Can you provide more details to reproduce this? And also if possible can you email me your stack details at vikram.kalta@contentstack.com?

Thanks

**Comment by vkalta** [comment]
@josephmarkus We have released v3.0.1 with this fix.

Thanks.
